### PR TITLE
fix: Android TV text inputs functionality

### PR DESCRIPTION
**Pull request by dlowder-salesforce**
Signed-off-by: Ventsislav Dimitrov <4097884+vdmtrv@users.noreply.github.com>

## Summary
Selecting an Android TV text input doesn't open the soft keyboard.
(This is merged from the original commit on the 0.62 release branch)

## Changelog

[Android TV] [FIXED] - Open soft keyboard upon selecting a text input

## Test Plan
* Create a simple view with a text input
* Attempt to enter text in the input
* Soft keyboard should appear and allow the user to enter desired text

### CODE REVIEW

**Review by vdmtrv — APPROVED** [review]

File: ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactEditText.java [hunk 1]
```diff
@@ -241,16 +265,22 @@ public boolean requestFocus(int direction, Rect previouslyFocusedRect) {
...
     return isFocused();
   }
 
   private boolean requestFocusInternal() {
     setFocusableInTouchMode(true);
-    // We must explicitly call this method on the super class; if we call requestFocus() without
-    // any arguments, it will call into the overridden requestFocus(int, Rect) above, which no-ops.
     boolean focused = super.requestFocus(View.FOCUS_DOWN, null);
-    if (getShowSoftInputOnFocus()) {
+    if (getShowSoftInputOnFocus() && !isTVDevice()) {
```
Comment (dlowder-salesforce) [inline ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactEditText.java:276]: I believe this is the cause of #129 .... @vdmtrv would it be ok to revert this?  Why was this added for TV devices?

File: ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactEditText.java [hunk 2]
```diff
@@ -241,16 +265,22 @@ public boolean requestFocus(int direction, Rect previouslyFocusedRect) {
...
       showSoftKeyboard();
+    } else {
+      if (isKeyboardOpened) {
```
Comment (EyMaddis) [inline ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactEditText.java:279]: super late to the party, but I think there is a `!` missing here...
